src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1432,7 +1432,13 @@ void primitive_inst::do_runtime_in_place_crop() {
 
     for (auto u : get_user_insts()) {
         if (u->get_node().is_type<crop>()) {
-            if (u->get_node().can_be_optimized()) {
+            // Check whether the crop is VariadicSplit op or not.
+            // The VariadicSplit shape infer requires executed value of input[2](split_lengths)
+            // So skip update_shape here when _node is input[2] of crop
+            auto crop_op_mode = u->_impl_params->typed_desc<crop>()->op_mode;
+            bool is_split_lengths = (crop_op_mode == cldnn::crop_ngraph_op_mode::variadic_split) &&
```

Comment:
crop has 3 inputs and do_runtime_in_place_crop is executed twice in previous 2 inputs before split_lenth node. but this condition only checks where current node is 3rd input or not.
Under this condition, isn't it possible that update_shape of crop is done before the split_length node is executed?

Reply:
Actually axis and split_lengths nodes are executed before input0. And axis should have valid number because it is mandatory to check crop buffer fusing in build time.

Reply:
Isn't it better to check user crop's 2nd input is non_constant?

Reply:
@yeonbok do_runtime_in_place_crop() is not called when node is constant. So checking non_constant here is not feasible.

Reply:
@yeonbok Updated as we discussed. Thanks!